
From: Hans Reiser <reiser@namesys.com>

Reiser4 is trying to add/remove pages to/from address space, so it needs 

remove_from_page_cache
__remove_from_page_cache
add_to_page_cache_lru

to be EXPORT_SYMBOL-ed.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/mm/filemap.c |    3 +++
 1 files changed, 3 insertions(+)

diff -puN mm/filemap.c~reiser4-export-remove_from_page_cache mm/filemap.c
--- 25/mm/filemap.c~reiser4-export-remove_from_page_cache	2004-12-27 00:30:51.255938472 -0800
+++ 25-akpm/mm/filemap.c	2004-12-27 00:30:51.262937408 -0800
@@ -118,6 +118,7 @@ void __remove_from_page_cache(struct pag
 	mapping->nrpages--;
 	pagecache_acct(-1);
 }
+EXPORT_SYMBOL(__remove_from_page_cache);
 
 void remove_from_page_cache(struct page *page)
 {
@@ -130,6 +131,7 @@ void remove_from_page_cache(struct page 
 	__remove_from_page_cache(page);
 	write_unlock_irq(&mapping->tree_lock);
 }
+EXPORT_SYMBOL(remove_from_page_cache);
 
 static int sync_page(void *word)
 {
@@ -375,6 +377,7 @@ int add_to_page_cache_lru(struct page *p
 		lru_cache_add(page);
 	return ret;
 }
+EXPORT_SYMBOL(add_to_page_cache_lru);
 
 /*
  * In order to wait for pages to become available there must be
_
